lsp/ARCHITECTURE.md

```diff
@@ -1,4 +1,18 @@
-# Architechture
+# Architecture
+
+The application consists of two components: the language server, and the language services.
+
+The **Language Server** is the containing application. It hosts language services, implements the Language server protocol (LSP), and queries information from language services in order to resepond to LSP requests.
+
+The **Language Services** are document-specific implementations that understand how to parse and interpret certain file types. These are used by the language server in order to provide LSP clients with details that power language assistance functionality.
+
+## Language Server
+
+![Langauge server architecture diagram](docs/images/language-server-architecture.svg)
```

Comment:
The diagram makes more sense. It seems like "service" is a concept we introduced? Calling it a "provider" would be less confusing. Then we would define them as:

```
The **Language Server** is implements the Language server protocol (LSP), and queries information from document-specific providers in order to resepond to LSP requests.

The **Language Providers** implement analysis of various document types. These are used by the language server to serve LSP client requests.
```

Reply:
Service is a concept we introduced so it's not mentioned anywhere in the LSP. It's because the [json](https://github.com/microsoft/vscode-json-languageservice) intellisense implementation is in a different repository than the actual server implementation so we started calling the json intellisense provider the language service and the thing that takes in the LSP requests and redirects them to the service as the language server. The YAML language server ran with that idea as well, except for it has the server + service implementation in the same repo. Other language servers don't have this kind of idea though. It's unique to JSON/YAML iirc

That's why in my previous comment I wondered if we should pick a different name, since quite a few stakeholders were getting confused between service and server

Reply:
> Service is a concept we introduced

Yup, just mentioning that little bit of info removes a lot of confusion here.